src/engine/tt_hash_table.h

```diff
@@ -14,20 +15,95 @@ enum TtHashFlag : uint8_t {
     TtHashBeta = 2,
 };
 
+/*
+ * Non-key members are encoded for compression. From lower bits to higher, entries are:
+ * depth: 8 bits, i.e. 0000 ... 0000 0000 1111 1111
+ * flag: 2 bits, i.e. 0000 ... 0000 0011 1000 0000
+ * score: 16 bits for value + 1 for sign (see board_defs.h)
+ * generation: 4 bits
+ * move: 26 bits (see move_types.h)
+ * */
+
+class TtHashEntryData {
+public:
+    explicit TtHashEntryData() = default;
+
+    explicit TtHashEntryData(uint64_t depth, uint64_t flag, int64_t score, uint64_t generation, uint64_t move)
+        : m_data(
+              depth
+              | flag << s_flagShift
+              // Most significant bit: sign (1 if negative)
+              | (score > 0 ? score : -score + (1 << 16)) << s_scoreShift
+              | generation << s_generationShift
+              | move << s_moveShift)
+    {
+    }
+
+    constexpr inline uint8_t getDepth() const
```

Comment:
constexpr is default inline. I added inline other places because I was not aware that it was default, so my bad :)

```suggestion
    constexpr uint8_t getDepth() const
```